libraries/AP_Scripting/examples/LED_snake.lua

```diff
@@ -0,0 +1,97 @@
+--[[
+script to send a 'rainbow snake' around a long set of LEDs on a single pin
+--]]
+
+local num_leds = 60
+local led_base = 0
+local snake_len = 5
+local brightness = 1
+
+--[[
+ use SERVOn_FUNCTION 94 for LED. We can control up to 16 separate strips of LEDs
+ by putting them on different channels
+--]]
+local chan = SRV_Channels:find_channel(94)
+
+if not chan then
+    gcs:send_text(6, "LEDs: channel not set")
+    return
+end
+
+-- find_channel returns 0 to 15, convert to 1 to 16
+chan = chan + 1
+
+gcs:send_text(6, "LEDs: chan=" .. tostring(chan) .. " nLEDs=" .. tostring(num_leds))
+
+-- initialisation code
+serialLED:set_num_LEDs(chan,  num_leds + led_base)
+
+-- constrain a value between limits
+function constrain(v, vmin, vmax)
+   if v < vmin then
+      v = vmin
```

Comment:
```suggestion
      return vmin
```

Reply:
No point in continuing as you've hit a limit, and saves you from doing extra checks/manipulations.